docs/content/en/docs/pipeline-stages/lifecycle-hooks.md

```diff
@@ -0,0 +1,116 @@
+          os: [darwin, linux]
+        - command: ["cmd.exe", "/C", "hook.bat"]
+          os: [windows]
+```
+This config snippet defines that `hook.sh` (for `darwin` or `linux` OS) or `hook.bat` (for `windows` OS) will be executed `before` and `after` each build for artifact `hooks-example`.
+
+### `before-sync` and `after-sync`
+
+Example: _skaffold.yaml_ snippet
+```yaml
+build:
+  artifacts:
+  - image: hooks-example
+    sync: 
+      auto: {}
+      hooks:
+        before:
+          - host:
+              command: ["sh", "-c", "./hook.sh"]
+              os: [darwin, linux]
+          - host:
+              command: ["cmd.exe", "/C", "hook.bat"]
+              os: [windows]
+        after:
+          - host:
+              command: ["sh", "-c", "./hook.sh"]
+              os: [darwin, linux]
+          - host:
+              command: ["cmd.exe", "/C", "hook.bat"]
+              os: [windows]
+```
+This config snippet defines that `hook.sh` (for `darwin` or `linux` OS) or `hook.bat` (for `windows` OS) will be executed `before` and `after` each file sync operation for artifact `hooks-example`.
+
+### `before-deploy` and `after-deploy` (_to be implemented_)
+
+### Environment variables
+
+The following environment variables will be available for the corresponding phase host hooks, that can be resolved in both inline commands or scripts.
+
+Environment variable | Description | Availability
+-- | -- | --
+$SKAFFOLD_IMAGE | The fully qualified image name. For example, “gcr.io/image1:tag” | Build, Sync
+$SKAFFOLD_PUSH_IMAGE | Set to true if the image in $IMAGE is expected to exist in a remote registry. Set to false if the image is expected to exist locally. | Build
+$SKAFFOLD_IMAGE_REPO | The image repo. For example, “gcr.io/image1” | Build
+$SKAFFOLD_IMAGE_TAG | The image tag. For example, “tag” | Build
+$SKAFFOLD_BUILD_CONTEXT | An absolute path to the directory this artifact is meant to be built from. Specified by artifact context in the skaffold.yaml. | Build
+$SKAFFOLD_FILES_ADDED_OR_MODIFIED | Semi-colon delimited list of absolute path to all files synced or to be synced in current dev loop that have been added or modified | Sync
+$SKAFFOLD_FILES_DELETED | Semi-colon delimited list of absolute path to all files synced or to be synced in current dev loop that have been deleted | Sync
+$SKAFFOLD_RUN_ID | Run specific UUID label for deployed or to be deployed resources | Deploy
+$SKAFFOLD_DEFAULT_REPO | The resolved default repository | All
+$SKAFFOLD_RPC_PORT | TCP port to expose event API | All
+$SKAFFOLD_HTTP_PORT | TCP port to expose event REST API over HTTP | All
+$SKAFFOLD_KUBE_CONTEXT | The resolved Kubernetes context | Sync, Deploy
+$SKAFFOLD_NAMESPACES | Comma separated list of Kubernetes namespaces | Sync, Deploy
+$SKAFFOLD_WORK_DIR | The workspace root directory | All
+Local environment variables | The current state of the local environment (e.g. $HOST, $PATH). Determined by the golang os.Environ function. | All
```

Comment:
Should this line be here?

Reply:
you're right. I think that's what was causing the weird scroll behavior also